Earthfile

```diff
@@ -39,241 +64,325 @@ setup:
-
+  RUN echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(lsb_release -cs) main" | tee /etc/apt/sources.list.d/ros2.list > /dev/null \
+    && apt update
+
+###############################################################################
+### Setup Stage
+# This stage is responsible for setting up the ROS 2 workspace and the required
+# dependencies for the workspace.
+###############################################################################
+setup:
+  FROM +pre-installation
   # Install required software development tools and ROS tools (and vim included for convenience)
-  RUN --mount=type=cache,mode=0777,target=/var/cache/apt,sharing=locked,id=cache_apt_cache \
-      --mount=type=cache,mode=0777,target=/var/lib/apt,sharing=locked,id=lib_apt_cache \
-      apt-get update && \
-      apt-get install -y \
-        bison \
-        build-essential \
-        cmake \
-        git \
-        python3-colcon-common-extensions \
-        python3-flake8 \
-        python3-flake8-blind-except \
-        python3-flake8-builtins \
-        python3-flake8-class-newline \
-        python3-flake8-comprehensions \
-        python3-flake8-deprecated \
-        python3-flake8-docstrings \
-        python3-flake8-import-order \
-        python3-flake8-quotes \
-        python3-pip \
-        python3-pytest \
-        python3-pytest-cov \
-        python3-pytest-repeat \
-        python3-pytest-rerunfailures \
-        python3-rosdep \
-        python3-rosinstall-generator \
-        python3-setuptools \
-        python3-vcstool \
-        wget \
-        vim
-
-  # Define the username and key variables
-  ENV USERNAME spaceros-user
-  ENV HOME_DIR=/home/${USERNAME}
-  ENV SPACEROS_DIR=/opt/spaceros
-
-  # Create the spaceros directory
-  RUN mkdir --mode=777 -p ${SPACEROS_DIR}
-
-  # Create a spaceros user
-  RUN useradd -m ${USERNAME} && \
-    echo "${USERNAME}:${USERNAME}" | chpasswd && \
-    usermod --shell /bin/bash ${USERNAME} && \
-    usermod -aG sudo ${USERNAME} && \
-    mkdir -p /etc/sudoers.d && \
-    echo "${USERNAME} ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/${USERNAME} && \
-    chmod 0440 /etc/sudoers.d/${USERNAME}
-  USER ${USERNAME}
-  WORKDIR ${SPACEROS_DIR}
-
-  # Update the OpenGL version
-  RUN sudo add-apt-repository ppa:kisak/kisak-mesa
-  RUN --mount=type=cache,mode=0777,target=/var/cache/apt,sharing=locked,id=cache_apt_cache \
-      --mount=type=cache,mode=0777,target=/var/lib/apt,sharing=locked,id=lib_apt_cache \
-      sudo apt update && sudo apt upgrade -y
+  RUN apt-get install -y python3-rosinstall-generator
 
-  # Create install location and copy in relevant scripts
-  COPY --chown ${USERNAME}:${USERNAME} --dir docker/scripts/ ${SPACEROS_DIR}
+  COPY scripts ./
```

Comment:
Actually getting into playing around with the dev image now! We need to keep the generate and merge scripts somewhere in the final image so that applications that build on this have access to them, as well.

Maybe in ${SPACEROS_DIR}/scripts or something? Somewhere else?

Reply:
c